### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by joyapisi**
## What I Have Done:
- Create a class Classroom
- Create the has-many/belongs-to relationship between Classroom and Student
- Create a class Book
- Create a class Rental
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './rental'
+
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (prolajumokeoni) [inline book.rb:10]: - [x] Kindly add a  method for adding rentals, see example below ⬇️  

```
def add_rental(person, date)
   Rental.new(date, self, person)
  end

```

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @joyapisi 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Associations setup 


## Required Changes ♻️

- Kindly refactor  your Person class please:
 
- [ ] ensure  Getters and Setters for @rentals is present, update your ``attr_accessor :name, :age, :rentals`` 


- [ ]  your rentals should be initialized `@rentals = []`
- [ ] add a method for adding rentals, The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association

  `def add_rental(book, date)
    Rental.new(date, book, self)
  end`


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @joyapisi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Association setup ✔️ 
- Professional readme file ✔️
- Commit messages meaningful ✔️

 ## Required Changes ♻️
_Check the comments under the review._
- [ ] As required by the previous reviewer, kindly complete the person class. To do this, there are few things you need to add, they are:
- Adding the getter and setter for @rentals is present and you can with the attribute assessor shorthand `attr_accessor :name, :age, :rentals`
- You will need to initialize the @rentals too like `@rentals = []`
- You will need to  add a method for adding rentals, the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. It would look like this:
```
def add_rental(book, date)
   Rental.new(date, book, self)
end
```
###nYour current code
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @joyapisi,

## APPROVED 🟢

## Highlights
✔️ Associations

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**